### PR TITLE
Expand more functions.

### LINKED ISSUE #3: Expand more functions which can query permission for the current login user.
In the last commit, I only completed the function of querying whether the current login user has specific permissions.I think we can expand more functions which can query permission for the current login user.

### PR DESCRIPTION
**Pull request by shy1st**
Fix: https://github.com/jcasbin/shiro-casbin/issues/3

### CODE REVIEW

File: src/main/java/org/casbin/shiro/authorize/rbac/program/Operation.java [hunk 1]
```diff
@@ -28,20 +30,114 @@
...
+        if (isLogin()) {
+            return false;
+        }
+        Subject subject = SecurityUtils.getSubject();
+        return e.enforce(subject.getPrincipal().toString(), act);
+    }
+
     /**
-     * hasAuthorization determines whether the current login user has a permission.
+     * hasAuthorizations determines whether the user currently logged on has a permission of the object.
```
Comment (hsluoyz) [inline src/main/java/org/casbin/shiro/authorize/rbac/program/Operation.java]: Trailing s?

File: src/main/java/org/casbin/shiro/authorize/rbac/program/Operation.java [hunk 2]
```diff
@@ -28,20 +30,114 @@
...
      *
-     * @param e the enforcer;
      * @param obj the object (usually means resources).
      * @param act the action (usually means the operation that the user performs on the resource).
      * @return whether the current login user has the permission.
      */
-    public static boolean hasAuthorization(Enforcer e, String obj, String act) {
-        if (!SecurityUtils.getSubject().isAuthenticated()) {
-            LogUtil.logPrintfError("Please login");
+    public boolean hasAuthorization(String obj, String act) {
```
Comment (hsluoyz) [inline src/main/java/org/casbin/shiro/authorize/rbac/program/Operation.java]: `hasAuthorization` is not correct English. Should be `isAuthorized` or `isPermitted` or `hasPermission`.

File: src/main/java/org/casbin/shiro/authorize/rbac/program/Operation.java [hunk 3]
```diff
@@ -28,20 +30,114 @@
...
+        if (isLogin()) {
             return false;
         }
         Subject subject = SecurityUtils.getSubject();
         return e.enforce(subject.getPrincipal().toString(), obj, act);
     }
+
+    /**
+     * hasAuthorizations determines whether the user currently logged on has a permission of the object.
```
Comment (hsluoyz) [inline src/main/java/org/casbin/shiro/authorize/rbac/program/Operation.java]: Trailing s?

File: src/main/java/org/casbin/shiro/authorize/rbac/program/Operation.java [hunk 4]
```diff
@@ -28,20 +30,114 @@
...
+     *
+     * @param dom the domain the roles belong to.
+     * @param obj the object (usually means resources).
+     * @param act the action (usually means the operation that the user performs on the resource).
+     * @return whether the current login user has the permission.
+     */
+    public boolean hasAuthorization(String dom, String obj, String act) {
```
Comment (hsluoyz) [inline src/main/java/org/casbin/shiro/authorize/rbac/program/Operation.java]: `hasAuthorization` is not correct English. Should be `isAuthorized` or `isPermitted` or `hasPermission`.

File: src/main/java/org/casbin/shiro/authorize/rbac/program/Operation.java [hunk 5]
```diff
@@ -28,20 +30,114 @@
...
+        return e.hasRoleForUser(subject.getPrincipal().toString(), role);
+    }
+
+
+    /**
+     * isLogin judge whether there is an user currently logged in.
+     *
+     * @return whether there is an user currently logged in.
+     */
+    private boolean isLogin() {
```
Comment (hsluoyz) [inline src/main/java/org/casbin/shiro/authorize/rbac/program/Operation.java]: `isLogin` is not correct English. Should be `isLoggedIn`

File: src/test/java/org/casbin/shiro/controller/rbac/PermitAnnotationController.java [hunk 6]
```diff
@@ -12,32 +12,22 @@
...
+package org.casbin.shiro.controller.rbac;
 
-import org.casbin.shiro.authorize.rbac.annotation.PreAuth;
+import org.casbin.shiro.authorize.rbac.annotation.auth.EnforcerAuth;
+import org.casbin.shiro.authorize.rbac.annotation.permit.EnforcerPermit;
 import org.springframework.web.bind.annotation.RestController;
 
 @RestController
-public class RbacAuthAnnotationController {
+public class PermitAnnotationController {
```
Comment (hsluoyz) [inline src/test/java/org/casbin/shiro/controller/rbac/PermitAnnotationController.java]: What does `Permit` here mean?

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: src/main/java/org/casbin/shiro/authorize/rbac/program/Operation.java (hunk 4 above)
Comment (shy1st) [reply src/main/java/org/casbin/shiro/authorize/rbac/program/Operation.java]: Ok,I will take your advice.

File: src/main/java/org/casbin/shiro/authorize/rbac/program/Operation.java (hunk 5 above)
Comment (shy1st) [reply src/main/java/org/casbin/shiro/authorize/rbac/program/Operation.java]: Ok,I will take your advice.

File: src/main/java/org/casbin/shiro/authorize/rbac/program/Operation.java (hunk 2 above)
Comment (shy1st) [reply src/main/java/org/casbin/shiro/authorize/rbac/program/Operation.java]: Ok,I will take your advice.

File: src/main/java/org/casbin/shiro/authorize/rbac/program/Operation.java (hunk 1 above)
Comment (shy1st) [reply src/main/java/org/casbin/shiro/authorize/rbac/program/Operation.java]: Maybe trailing s is wrong.I will take your advice.

File: src/main/java/org/casbin/shiro/authorize/rbac/program/Operation.java (hunk 3 above)
Comment (shy1st) [reply src/main/java/org/casbin/shiro/authorize/rbac/program/Operation.java]: Maybe trailing s is wrong.I will take your advice.

File: src/test/java/org/casbin/shiro/controller/rbac/PermitAnnotationController.java (hunk 6 above)
Comment (shy1st) [reply src/test/java/org/casbin/shiro/controller/rbac/PermitAnnotationController.java]: I create a annotation name EnforcerPermit, This is the controller for test the annotation.

Comment (hsluoyz) [reply src/test/java/org/casbin/shiro/controller/rbac/PermitAnnotationController.java]: OK

**Comment by shy1st** [comment]
In this commit, I rewrite the logic of enforceauth and allow users to configure datasource in the configuration file. At present, the enforceauth can judge whether the current user has the right to access a specific interface,like this:
捕获

**Comment by hsluoyz** [comment]
@shy1st plz fix CI:

https://travis-ci.com/github/jcasbin/shiro-casbin/jobs/476866261

**Review by hsluoyz — APPROVED** [review]

**Comment by hsluoyz** [comment]
lgtm

**Pull request merged**